src/components/Steps/StepsDot.jsx

```diff
@@ -0,0 +1,53 @@
+import cx from "classnames";
+import NOOP from "lodash/noop";
+import PropTypes from "prop-types";
+import { useMemo } from "react";
+import { STEPS_CSS_BASE_CLASS } from "./StepsConstants";
+import { BEMClass } from "../../helpers/bem-helper";
+import { useChangeStepFunction } from "./hooks/useChangeStepFunction";
+
+const CSS_BASE_CLASS = `${STEPS_CSS_BASE_CLASS}-header`;
+const bemHelper = BEMClass(CSS_BASE_CLASS);
+
+export const StepsDot = ({
+  stepsCount,
+  onChangeActiveStep,
+  activeStepIndex,
+  stepIndex,
+  ariaCurrent,
+  stepDescriptionFunc
+}) => {
+  const isActive = activeStepIndex === stepIndex;
```

Comment:
please pass it in the props there is no need to rerender all on every activeStepIndex

like in the parent component pass - `isActive={activeStepIndex === stepIndex}`